File: notes_and_examples/assertion_example.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.junit.jupiter.api.*;

public class GoogleTitleTest {

    WebDriver driver;

    @BeforeEach
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "path_to_chromedriver"); // Set the path to your chromedriver
        driver = new ChromeDriver();
    }

    @Test
    public void testGooglePageTitle() {
        driver.get("https://www.google.com");
        String expectedTitle = "Google";
        String actualTitle = driver.getTitle();
        // Assert whether the expected title matches the actual title
        Assertions.assertEquals(expectedTitle, actualTitle, "Expected title '" + expectedTitle + "' but got '" + actualTitle + "'");
    }

    @AfterEach
    public void tearDown() {
        driver.quit();
    }
}

File: notes_and_examples/assertion_example.py
import unittest
from selenium import webdriver # pip install selenium

class GoogleTitleTest(unittest.TestCase):

    def setUp(self):
        self.driver = webdriver.Chrome()  # You'll need to set the appropriate webdriver (e.g., Chrome, Firefox)

    def test_google_page_title(self):
        self.driver.get('https://www.google.com')
        expected_title = "Google"
        actual_title = self.driver.title
        # Assert whether the expected title matches the actual title
        self.assertEqual(expected_title, actual_title, f"Expected title '{expected_title}' but got '{actual_title}'")

    def tearDown(self):
        self.driver.quit()

if __name__ == '__main__':
    unittest.main()


"""

This test case navigates to Google's homepage and checks whether the title of the page is "Google." 
If the page title doesn't match "Google," the test will fail, indicating that the assertion has failed. 

"""
